qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
+        {
+            (0,): InstructionProperties(error=.001, duration=5e-5),
+            (1,): InstructionProperties(error=.002, duration=6e-5),
+            (2,): InstructionProperties(error=.2, duration=5e-7)
+        }
+    )
+    print(target)
+
+.. parsed-literal::
+
+    Target
+    Number of qubits: 3
+    Instructions:
+        cx
+            (0, 1):
+                Duration: 5e-07 sec.
+                Error Rate: 0.0001
+        u
+            (0,):
+                Duration: 5e-08 sec.
+                Error Rate: 1e-05
+            (1,):
+                Duration: 6e-08 sec.
+                Error Rate: 2e-05
+        rz
+            (1,):
+                Duration: 5e-08 sec.
+                Error Rate: 1e-05
+            (2,):
+                Duration: 6e-08 sec.
+                Error Rate: 2e-05
+        ry
+            (1,):
+                Duration: 5e-08 sec.
+                Error Rate: 1e-05
+            (2,):
+                Duration: 6e-08 sec.
+                Error Rate: 2e-05
+        rx
+            (1,):
+                Duration: 5e-08 sec.
+                Error Rate: 1e-05
+            (2,):
+                Duration: 6e-08 sec.
+                Error Rate: 2e-05
+        cz
+            (1, 2):
+                Duration: 5e-07 sec.
+                Error Rate: 0.0001
+            (2, 0):
+                Duration: 5e-07 sec.
+                Error Rate: 0.0001
+        measure
+            (0,):
+                Duration: 5e-05 sec.
+                Error Rate: 0.001
+            (1,):
+                Duration: 6e-05 sec.
+                Error Rate: 0.002
+            (2,):
+                Duration: 5e-07 sec.
+                Error Rate: 0.2
+
+This :class:`~.Target` represents a 3 qubit QPU that supports :class:`~.CXGate` between qubits
+0 and 1, :class:`~.UGate` on qubits 0 and 1, :class:`~.RZGate`, :class:`~.RXGate`,
+and :class:`~.RYGate` on qubits 1 and 2, :class:`~.CZGate` between qubits 1 and 2, and qubits
+2 and 0, and :class:`~.Measure` on all qubits.
+
+There are also specific data structures to represent a specific subset of information from the
+:class:`~.Target`. For example, the :class:`~.CouplingMap` class is used to solely represent the
+connectivity constraints of a backend as a directed graph. A coupling map can be generated from
+a :class:`~.Target` using the :meth:`.Target.build_coupling_map` method. These data structures
+typically pre-date the :class:`~.Target` class but are often used with transpiler passes that do
+not work natively with a :class:`~.Target` instance or with backends that aren't using the latest
+:class:`~.BackendV2` interface. However, they do also offer more targeted interaction methods
+with the limited view they provide. For example, :class:`~.CouplingMap` provides methods for
+analyzing the connectivity graph, such as :meth:`.CouplingMap.shortest_undirected_path`, which
+can be useful if analysis of the connectivity constraints of a backend is desired.
```

Comment:
Yeah, to be more future facing I can reword this to be more about backwards compatibility. I did it this mainly based on #9263 where I was just using `CouplingMap` internally still because we're still effectively working with a graph for all of layout and routing. But as part of #7812 we'll have to add these kind of methods directly on the `Target` (and likely model things as hyper graphs).